contracts/extensions/ClaimExtension.sol

```diff
@@ -0,0 +1,690 @@
+     * @param _airdropSettings      Struct of airdrop setting for Set including accepted airdrops, feeRecipient,
+     *                              airdropFee, and indicating if anyone can call an absorb
+     */
+    function initializeAirdropModule(
+        IDelegatedManager _delegatedManager,
+        IAirdropModule.AirdropSettings memory _airdropSettings
+    )
+        external
+        onlyOwnerAndValidManager(_delegatedManager)
+    {
+        _initializeAirdropModule(_delegatedManager.setToken(), _delegatedManager, _airdropSettings);
+    }
+
+    /**
+     * ONLY OWNER: Initializes ClaimModule on the SetToken associated with the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize the ClaimModule for
+     * @param _anyoneClaim          Boolean indicating if anyone can claim or just manager
+     * @param _rewardPools          Addresses of rewardPools that identifies the contract governing claims. Maps to same index integrationNames
+     * @param _integrationNames     Human-readable names matching adapter used to collect claim on pool. Maps to same index in rewardPools
+     */
+    function initializeClaimModule(
+        IDelegatedManager _delegatedManager,
+        bool _anyoneClaim,
+        address[] calldata _rewardPools,
+        string[] calldata _integrationNames
+    )
+        external
+        onlyOwnerAndValidManager(_delegatedManager)
+    {
+        _initializeClaimModule(_delegatedManager.setToken(), _delegatedManager, _anyoneClaim, _rewardPools, _integrationNames);
+    }
+
+    /**
+     * ONLY OWNER: Initializes ClaimExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external onlyOwnerAndValidManager(_delegatedManager) {
+        ISetToken setToken = _delegatedManager.setToken();
+
+        _initializeExtension(setToken, _delegatedManager);
+
+        emit ClaimExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Initializes ClaimExtension to the DelegatedManager and AirdropModule and ClaimModule to the SetToken
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     * @param _airdropSettings      Struct of airdrop setting for Set including accepted airdrops, feeRecipient,
+     *                              airdropFee, and indicating if anyone can call an absorb
+     * @param _anyoneClaim          Boolean indicating if anyone can claim or just manager
+     * @param _rewardPools          Addresses of rewardPools that identifies the contract governing claims. Maps to same index integrationNames
+     * @param _integrationNames     Human-readable names matching adapter used to collect claim on pool. Maps to same index in rewardPools
+     */
+    function initializeModulesAndExtension(
```

Comment:
Just noting as part of review that all the parameter validations are executed at the module layer